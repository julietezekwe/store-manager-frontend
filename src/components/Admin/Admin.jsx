import React from 'react';

const Admin = () => (
        <div className="dashboard myCard col-2">
          <h3> Dashboard</h3>
          <p className="active"><a href="/admin/">Products</a></p>
          <hr />
          <p className="active"><a href="/admin/sales">Sales Records</a></p>
          <hr />
          <p className="active"><a href='/admin/users'>Users</a></p>
        </div>
);

export default Admin;
