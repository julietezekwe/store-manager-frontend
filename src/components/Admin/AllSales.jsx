import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAllSales } from '../../actions/salesAction';

export class AllSales extends Component {
  async componentDidMount() {
    await this.props.getAllSales();
  }

  render() {
    const { sales } = this.props.sales;
    return (
      <React.Fragment>
        <div className="col-8">
          <div className="center">
            <div className="col-3 " id="totalProducts">
              Total Products Sold: {sales.totalQuantitySold}
            </div>
            <div className="col-4" id="totalPrice">
              Amount Realized: &#8358; {sales.totalAmount}
            </div>
            <div className="col-3" id="totalSales">
              Unique Products No: {sales.totalProductsSold}
            </div>
          </div>

          <div className="card col-11">
            <table className="table">
              <thead>
                <tr>
                  <th>Product Id</th>
                  <th>Product Name</th>
                  <th>Description</th>
                  <th>Quantity</th>
                  <th>Price</th>
                  <th>Seller</th>
                </tr>
              </thead>
              <tbody>
                {sales.SalesModel
                  && sales.SalesModel.map((sale, i) => (
                    <tr key={i}>
                      <td>{sale.productid}</td>
                      <td>{sale.productname}</td>
                      <td>{sale.description}</td>
                      <td>{sale.quantity}</td>
                      <td>{sale.price}</td>
                      <td>{sale.name}</td>
                    </tr>
                  ))}
              </tbody>
            </table>
          </div>
        </div>
      </React.Fragment>
    );
  }
}
const mapStateToProps = state => ({
  sales: state.sales,
});
export default connect(
  mapStateToProps,
  { getAllSales },
)(AllSales);
