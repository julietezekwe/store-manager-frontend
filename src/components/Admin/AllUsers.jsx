import React, { Component } from 'react';
import { connect } from 'react-redux';
import Form from '../common/Form.jsx';
import { getAllUsers } from '../../actions/usersAction';

export class AllUsers extends Component {
  async componentDidMount() {
    await this.props.getAllUsers();
  }

  render() {
    const { users, isLoadingUsers } = this.props.login;

    return (
      <React.Fragment>
        <div className="col-8">
          <Form action="addUser" />
          <div className="card col-11">
            <table className="table">
              <thead>
                <tr>
                  <th>Name</th>
                  <th>Username</th>
                  <th>Email</th>
                  <th>Role</th>
                  <th>Action</th>
                </tr>
              </thead>
              <tbody>
                {!isLoadingUsers
                  && users.UsersModel.map(user => (
                    <tr key={user.id}>
                      <td>{user.name}</td>
                      <td>{user.username}</td>
                      <td>{user.email}</td>
                      <td>{user.role}</td>
                      <td>
                        <Form action="eidtUser" userId={user.id} /> &nbsp;
                        &nbsp;
                        <Form action="deleteUser" userId={user.id} />
                      </td>
                    </tr>
                  ))}
              </tbody>
            </table>
          </div>
        </div>
      </React.Fragment>
    );
  }
}
const mapStateToProps = state => ({
  login: state.login,
  users: state.login.users,
});
export default connect(
  mapStateToProps,
  { getAllUsers },
)(AllUsers);
