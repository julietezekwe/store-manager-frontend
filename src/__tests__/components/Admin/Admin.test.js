/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import Admin from '../../../components/Admin/Admin.jsx';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Admin />);
});
describe('App Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should display three links on the dashboard', () => {
    expect(wrapper.find('a')).toHaveLength(3);
  });
  it('should have the right path links for the hrefs', () => {
    expect(wrapper.find('a').at(0).prop('href')).toEqual('/admin/');
  });
});
