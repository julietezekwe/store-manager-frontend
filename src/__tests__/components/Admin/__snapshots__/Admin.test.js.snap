// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<div
  className="dashboard myCard col-2"
>
  <h3>
     Dashboard
  </h3>
  <p
    className="active"
  >
    <a
      href="/admin/"
    >
      Products
    </a>
  </p>
  <hr />
  <p
    className="active"
  >
    <a
      href="/admin/sales"
    >
      Sales Records
    </a>
  </p>
  <hr />
  <p
    className="active"
  >
    <a
      href="/admin/users"
    >
      Users
    </a>
  </p>
</div>
`;
